docs/testing/cypress.mdx

```diff
@@ -0,0 +1,191 @@
+2. In the navigation sidebar, select **API Keys**.
+3. In the **Quick Copy** section, copy your Clerk publishable and secret key.
+
+> [!WARNING]
+> Ensure you provide the secret key in a secure manner, to avoid leaking it to third parties. For example, if you are using GitHub Actions, refer to [*Using secrets in GitHub Actions*](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions).
+
+
+### Global setup
+
+To set up Clerk with Cypress, add the following code to your [`cypress.config.ts`](https://docs.cypress.io/guides/references/configuration) file.
+
+```tsx filename="cypress.config.ts"
+import { clerkSetup } from '@clerk/testing/cypress';
+import { defineConfig } from 'cypress';
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
+      return clerkSetup({ config });
+    },
+    baseUrl: 'http://localhost:3000' // your app's URL
+  }
+});
+```
+
+`clerkSetup` will obtain a [Testing Token](/docs/testing/overview#testing-tokens) when your test suite starts, making it available for all subsequent tests to use.
+
+### Use Clerk Testing Tokens
+
+Now that Cypress is configured with Clerk, you can use the `setupClerkTestingToken()` function in your tests to augment them with the Testing Token. See the following example:
+
+```tsx filename="my-test.cy.ts"
+import { setupClerkTestingToken } from "@clerk/testing/cypress";
+
+it("sign up", () => {
+  setupClerkTestingToken();
+
+  cy.visit('/sign-up');
+  // ...
+});
+```
+
+</Steps>
+
+## Custom Commands
+
+The `@clerk/testing` package also provides [Cypress custom commands](https://docs.cypress.io/api/cypress-api/custom-commands) to sign in/sign out with Clerk in your Cypress tests without having to interact with the UI.
+To use these commands, you must import them in your `cypress/support/commands.ts` file.
+
+Example:
+```ts {{ filename: 'cypress/support/commands.ts' }}
+/// <reference types="cypress" />
+import { addClerkCommands } from '@clerk/testing/cypress';
+addClerkCommands({ Cypress, cy });
+
+export {};
+```
+
+### `cy.clerkSignIn`
+
+The `cy.clerkSignIn` command is used to sign in a user using Clerk. This custom command supports only the following first factor strategies:
+
+- Password
+- Phone code
+- Email code
+
+**Note:** Multi-factor authentication is not supported.
+
+> [!NOTE]
+> This helper internally uses the `setupClerkTestingToken`, so you don't need to call it separately.
+
+**Prerequisites**
+
+- You must call `cy.visit` before calling this command.
+- Navigate to a non-protected page that loads Clerk before using this command.
+
+**Parameters**
+
+This command accepts an object with the following properties:
+
+- `strategy`: The sign-in strategy. Supported strategies are `'password'`, `'phone_code'`, or `'email_code'`.
+- `identifier`: The user's identifier. This could be a username, a phone number, or an email.
+- `password`: The user's password. This is required only if the strategy is `'password'`.
```

Comment:
```suggestion
#### Parameters

This command accepts an object with the following properties:

- `strategy`: The sign-in strategy. Supported strategies are `'password'`, `'phone_code'`, or `'email_code'`.
    - `password`: The command will sign in the user using the provided password and identifier.
    - `phone_code`: You must have a user with a test phone number as an identifier (e.g., `+15555550100`).
    - `email_code`: You must have a user with a test email as an identifier (e.g., `your_email+clerk_test@example.com`).
- `identifier`: The user's identifier. This could be a username, a phone number, or an email.
- `password`: The user's password. This is required only if the `strategy` is set to `'password'`.
```